### PR TITLE
[test] Fix Slurm constraint check for cabinets

### PR DESCRIPTION
**Pull request by vkarak**
I also removed unnecessary programming environments from `valid_prog_environs`.

Fixes UES-859.

### CODE REVIEW

File: cscs-checks/system/slurm/slurm.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ def __init__(self):
                               'kesch:cn', 'kesch:pn',
                               'arolla:cn', 'arolla:pn',
                               'tsa:cn', 'tsa:pn']
-        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu']
+        self.valid_prog_environs = ['PrgEnv-cray']
```
Comment (teojgo) [inline cscs-checks/system/slurm/slurm.py:19]:
```suggestion
        self.valid_prog_environs = ['builtin']
```

File: cscs-checks/system/slurm/slurm.py [hunk 2]
```diff
@@ -35,7 +35,7 @@ def __init__(self):
...
```
Comment (teojgo) [inline cscs-checks/system/slurm/slurm.py:38]:
```suggestion
        self.valid_prog_environs = ['builtin']
```

**Comment by vkarak** [comment]
@teojgo I had the same thought, but I was reluctant to do it, because some tests do have a build step, so it's better not to rely on the `builtin` environment. But I have not a strong feeling about this. I could use `builtin` for the run-only tests and `PrgEnv-cray` for the rest.

**Comment by teojgo** [comment]
> @teojgo I had the same thought, but I was reluctant to do it, because some tests do have a build step, so it's better not to rely on the `builtin` environment. But I have not a strong feeling about this. I could use `builtin` for the run-only tests and `PrgEnv-cray` for the rest.

Ah ok, then it is fine for me.

**Comment by vkarak** [comment]
CI is blocking due to #1298.

**Pull request merged**